#167765385 Swagger API documentation setup

voke1 opened the pull request:
### What does this PR do?

- Setup swagger API documentation

### Description of Task to be completed?

- Install and import `swagger-ui-express`
- Add swaggerDoc.json file to root directory.

### How should this be manually tested?

- Open `http://localhost:3000/api-docs/` in browser

### Any background context you want to provide?

-`Request` and `User` endpoint is set in the documentation for sample purposes

### What are the relevant pivotal tracker stories?

- #167765385

### Implementation Plan
- https://drive.google.com/file/d/1rox9mhsOGVyZDt4NKuR8tAT2TtJTgrUk/view?usp=sharing




![Screenshot-api-documention](https://user-images.githubusercontent.com/24835262/63135818-2b751000-bfc7-11e9-90bd-fc8727e7aa34.JPG)
![screenshot-ap-documentation-2](https://user-images.githubusercontent.com/24835262/63135931-a76f5800-bfc7-11e9-8c19-8dbc4ae5aa2d.JPG)

voke1 committed:
chore(index.js): setup swagger api documentation

voke1 committed:
chore(index.js): ensure proper commenting

voke1 committed:
chore(index.js) refactor code for efficiency

debelistic reviewed (approved):
Great job @voke1, well done

voke1 commented:
Thanks @debelistic

uggoprince reviewed (approved):
@voke1 it's well configured :+1:

did you use yml files or code comments?

voke1 commented:
> @voke1 it's well configured 👍
> 
> did you use yml files or code comments?

I used a json file. Swagger gives the option to use either yaml or json file. You will need to convert yaml file to json format.

voke1 committed:
chore(package-lock.json): regenerate package-lock.json file

voke1 committed:
Merge branch 'develop' into ch-swagger-doc-setup-#167765385

victor-nach committed:
chore(docker) - setup docker for backend

- Create a docker file that would create an image to run the application
- Create a docker-compose.yml file that would run the docker file

[Finishes - #167765386]

voke1 committed:
chore(index.js): add line spacing in import statement

voke1 committed:
Merge branch 'develop' into ch-swagger-doc-setup-#167765385

voke1 committed:
Merge branch 'develop' into ch-swagger-doc-setup-#167765385

anneKay commented on src/index.js:
```diff
@@ -10,11 +10,17 @@ const fs = require("fs"),
     mongoose = require("mongoose");
 
+import {serve, setup} from 'swagger-ui-express';
+import swaggerDoc from '../swaggerDoc';
```
Please update this PR with develop.

voke1 replied:
> Please update this PR with develop.

Done

voke1 committed:
Merge branch 'develop' of https://github.com/andela/phoenix-bn-backend into develop

voke1 committed:
Merge branch 'develop' into ch-swagger-doc-setup-#167765385

voke1 committed:
chore(index.js): reorder import statement

anneKay commented on Dockerfile:
```diff
@@ -0,0 +1,14 @@
+RUN npm install --quiet
+
+# copy all the app's files from the project's root to our working directory
+COPY . .
```
Add newline here

voke1 replied:
Done

anneKay commented on docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+      # create user postgres as the default user for postgreSQL
+      - POSTGRES_USER
+      # create the default database as todos
+      - POSTGRES_DB
```
Add newline here

voke1 replied:
Done

anneKay commented on swaggerDoc.json:
```diff
@@ -0,0 +1,205 @@
+            "url": "https://github.com/andela/phoenix-bn-backend"
+        }
+    }
+}
```
Add new line here

voke1 replied:
Done

voke1 committed:
chore [Finishes - #167765385]: add new line

anneKay commented:
Good job @voke1, did you follow the routes we already in the code for this documentation. I noticed you have an `api/vi/request` and `api/vi/auth/signin` Do we have such paths in the codebase?

voke1 commented:
HI @anneKay. Not yet, we don't have any path in the codebase already. The two routes included are just for test purposes. They can be deleted or modified when we commence with documenting of endpoints. do you advise I remove them?

anneKay commented:
Please look in routes/api/users.js` file to see the routes we have

voke1 commented:
Oh Sorry, I didn't follow the routes we have already in the codebase. Do you mean I document these routes also? It was concluded in previous meeting that anyone working on a route should also add documentation for the route. Please advise.

anneKay commented:
In that case we can leave it as it is, let's consider the routes you added test routes for setting up the documentation
